cpp/src/arrow/util/string_test.cc

```diff
@@ -35,5 +37,12 @@ TEST(Trim, Basics) {
   }
 }
 
+TEST(ParseHexValue, Invalid) {
+  std::string input = "xy";
+  uint8_t output;
+  
+  ASSERT_RAISES(Invalid, ParseHexValue(input.c_str(), &output));
```

Comment:
Please also test with an invalid string such as `"@?"`.
Also, while you're at it, add a couple tests for valid values.

Reply:
@pitrou Thanks a lot for your valuable comments.
My previous fix was not complete. I have fixed it, and added both positive and negative test cases.
Please take a look. Thank you in advance.